packages/apify/src/actor.ts

```diff
@@ -1,458 +1,1320 @@
+     * that contains details about the actor run and its output (if any).
+     *
+     * Note that an actor task is a saved input configuration and options for an actor.
+     * If you want to run an actor directly rather than an actor task, please use the
+     * {@link Actor.call} function instead.
+     *
+     * For more information about actor tasks, read the [documentation](https://docs.apify.com/tasks).
+     *
+     * **Example usage:**
+     *
+     * ```javascript
+     * const run = await Actor.callTask('bob/some-task');
+     * console.log(`Received message: ${run.output.body.message}`);
+     * ```
+     *
+     * Internally, the `callTask()` function calls the
+     * [Run task](https://apify.com/docs/api/v2#/reference/actor-tasks/run-collection/run-task)
+     * and several other API endpoints to obtain the output.
+     *
+     * @param taskId
+     *  Allowed formats are `username/task-name`, `userId/task-name` or task ID.
+     * @param [input]
+     *  Input overrides for the actor task. If it is an object, it will be stringified to
+     *  JSON and its content type set to `application/json; charset=utf-8`.
+     *  Provided input will be merged with actor task input.
+     * @param [options]
      */
-    actorTaskId: string | null;
+    async callTask(taskId: string, input?: Dictionary, options: CallTaskOptions = {}): Promise<ActorRunWithOutput> {
+        const { token, ...rest } = options;
+        const client = token ? this.newClient({ token }) : this.apifyClient;
+
+        return client.task(taskId).call(input, rest);
+    }
 
     /**
-     * ID of the user who started the actor - note that it might be
-     * different than the owner ofthe actor (APIFY_USER_ID)
+     * Transforms this actor run to an actor run of a given actor. The system stops the current container and starts
+     * the new container instead. All the default storages are preserved and the new input is stored under the `INPUT-METAMORPH-1` key
+     * in the same default key-value store.
+     *
+     * @param targetActorId
+     *  Either `username/actor-name` or actor ID of an actor to which we want to metamorph.
+     * @param [input]
+     *  Input for the actor. If it is an object, it will be stringified to
+     *  JSON and its content type set to `application/json; charset=utf-8`.
+     *  Otherwise, the `options.contentType` parameter must be provided.
+     * @param [options]
      */
-    userId: string | null;
+    async metamorph(targetActorId: string, input?: unknown, options: MetamorphOptions = {}): Promise<void> {
+        if (!this.isAtHome()) {
+            log.warning('Actor.metamorph() is only supported when running on the Apify platform.');
+            return;
+        }
+
+        const {
+            customAfterSleepMillis = this.config.get('metamorphAfterSleepMillis'),
+            ...metamorphOpts
+        } = options;
+        const runId = this.config.get('actorRunId')!;
```

Comment:
Probably a nit for the future but we should probably have an object/enum for these keys, so we can strictly type them

Reply:
what you mean? config is already strictly typed, it both checks the keys as well as maps the return type based on it